spot-electron/src/application-window/applicationwindow.js

```diff
@@ -35,26 +39,55 @@ function createApplicationWindow() {
     });
 
     // Set event handlers
+    onlineDetector.addListener(OnlineDetector.ONLINE_STATUS_CHANGED, isOnline => {
+        clearTimeout(showCrashPageTimeout);
+
+        if (isOnline) {
+            loadDefaultUrl(applicationWindow);
+        } else {
+            applicationWindow.loadFile('src/static/offline.html');
+        }
+    });
+
     applicationWindow.webContents.on('crashed', () => {
+        if (!onlineDetector.getLastOnlineStatus()) {
+            return;
+        }
+
         // The setTimeout 0 is necessary clear the current stack or else
         // navigation will cause the main app to crash as well.
         setTimeout(() => {
             applicationWindow.loadFile('src/static/crashed.html')
                 .then(() => {
-                    setTimeout(() => {
-                        applicationWindow.loadURL(defaultSpotURL);
-                    }, 5000);
+                    showCrashPageTimeout = setTimeout(() =>
+                        loadDefaultUrl(applicationWindow), 5000);
                 });
         });
     });
 
-    applicationWindow.loadURL(defaultSpotURL, {
-        userAgent: `${applicationWindow.webContents.getUserAgent()} ${SPOT_ELECTRON_FEATURE_VERSION}`
+    applicationWindow.on('closed', () => {
+        onlineDetector.destroy();
     });
 
+    loadDefaultUrl(applicationWindow);
+
+    onlineDetector.start();
     logger.info(`Spot started with Spot-TV URL ${defaultSpotURL}`);
 }
 
+/**
+ * Loads the root Spot-TV url on the provided window.
+ *
+ * @param {BrowserWindow} browserWindow - The BrowserWindow instance in which
+ * the Spot-TV url should be opened.
+ * @returns {void}
+ */
+function loadDefaultUrl(browserWindow) {
+    browserWindow.loadURL(defaultSpotURL, {
+        userAgent: `${browserWindow.webContents.getUserAgent()} ${SPOT_ELECTRON_FEATURE_VERSION}`
```

Comment:
I think this may be adding the spot electron version on each loadURL. I remember this because I've tried to make a method like this in the past. Maybe worth check if that's not the case?

Reply:
You were right that it was appending the version multiple times. I made a change to set the user agent outside of url load.